htdocs/compta/facture/class/api_invoices.class.php

```diff
@@ -1414,6 +1415,20 @@ public function addPayment($id, $datepaye, $paymentid, $closepaidinvoices, $acco
 		$totaldeposits = $this->invoice->getSumDepositsUsed();
 		$resteapayer = price2num($this->invoice->total_ttc - $totalpaid - $totalcreditnotes - $totaldeposits, 'MT');
 
+		// hook to finalize the remaining amount, considering e.g. cash discount agreements
+		$parameters = array('$totalpaid'=>$totalpaid, '$totalcreditnotes'=>$totalcreditnotes, '$totaldeposits'=>$totaldeposits, 'remaintopay'=>$resteapayer);
+		$action = 'API_CUSTOMER_INVOICE';
```

Comment:
$action is commonly used with value
'' or 'read' for no action done, 'delete' or 'add', or 'update'.
Having its value to API_... Looks strange.
Do you really need this value. Can't we have it set to a common alteady used value ?

Reply:
Thanks for the comment. $action is not needed in the hooked function, therefore I now omitted it. What I did was replicating some (bad) example I saw somewhere else in the code, where a non existing $action was created this way for a hook.

In the long run I think it makes sense to add a new member function getRemainToPay() like the existing  getSommePaiement(), getSumCreditNotesUsed() or getSumDepositsUsed() functions, since the calculation ot the remainToPay happens in many places. So all the new hooks could then go into one place within the member function and overall the code would become a bit cleaner. I would keep this in mind for the time after v18.

Reply:
P.S. the travis CI problem seems to be unrelated to my code